src/App.js

```diff
@@ -96,6 +97,17 @@ class App {
 					$contentWrapper.html( this.api.getReplacementHtml() );
 					$( 'body' ).append( this.revisionPopup.$element );
 					this.attachContentListeners();
+
+					// Scroll down to the content if it is below or in the lower third of the
+					// viewport, e.g. on a diff page.
+					viewBottom = window.scrollY + ( window.innerHeight * ( 2 / 3 ) );
+					contentTop = $contentWrapper.offset().top;
+					infobarHeight = this.widget.$element.outerHeight( true );
+					if ( contentTop >= viewBottom ) {
```

Comment:
Just an idea here, but from experience with rcfilters, scrolling is a bit awkward/disruptive, we might want to consider a buffer (or threshhold) where if you're already almost at the article we don't scroll, and we only scroll if it makes sense to. This will prevent cases where you can see the article halfway up the screen and we scroll you like 10 pixels down, which is weird, but will keep the behavior of scrolling when the article is not visible or obviously there.

Also, we might want to use jquery's `.animate()` so the scrolling is slightly less janky.

See example here with the `min: 20` https://github.com/wikimedia/mediawiki/blob/master/resources/src/mediawiki.rcfilters/ui/FilterTagMultiselectWidget.js#L415

The actual scrolling method: https://github.com/wikimedia/mediawiki/blob/master/resources/src/mediawiki.rcfilters/ui/FilterTagMultiselectWidget.js#L819

Reply:
I like the animation idea, that'd help people realise what's happening.

This is already only scrolling if the target is out of or almost out of view. The threshold is currently a third of the viewport height, but could be smaller (i.e. a shorter area at the bottom of the screen).

@prtksxna do you have an preference?

Reply:
I missed the 2/3 calculation, good thinking!

Reply:
I've added the animation.